File: src/main/java/net/nextpulse/jacumulus/responses/ListInvoicesResponse.java
package net.nextpulse.jacumulus.responses;

import net.nextpulse.jacumulus.models.Invoice;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

/**
 * Response for the list invoices request
 *
 * @see <a href="https://apidoc.sielsystems.nl/content/contact-list-incoming-invoices">the api docs for incoming</a>
 * @see <a href="https://apidoc.sielsystems.nl/content/contact-list-outgoing-invoices">the api docs for outgoing</a>
 */
@XmlRootElement(name = "response")
public class ListInvoicesResponse extends AcumulusResponse {
  @XmlElement(name = "contact")
  private List<Invoice> invoices = new ArrayList<Invoice>();
  
  public List<Invoice> getInvoices() {
    return invoices;
  }
  
  public void setInvoices(List<Invoice> invoices) {
    this.invoices = invoices;
  }
}

File: src/main/java/net/nextpulse/jacumulus/responses/ListInvoicesResponse.kt
package net.nextpulse.jacumulus.responses

import net.nextpulse.jacumulus.responses.models.Invoice
import java.util.*
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlElementWrapper
import javax.xml.bind.annotation.XmlRootElement

/**
 * Response for the list invoices request

 * @see [the api docs for incoming](https://apidoc.sielsystems.nl/content/contact-list-incoming-invoices)

 * @see [the api docs for outgoing](https://apidoc.sielsystems.nl/content/contact-list-outgoing-invoices)
 */
@XmlRootElement(name = "response")
data class ListInvoicesResponse(
        @get:XmlElementWrapper(name = "contact")
        @get:XmlElement(name = "invoice")
        var invoices: List<Invoice> = ArrayList()
) : AcumulusResponse()